### PR TITLE
Add support to ballerina enum type

### LINKED ISSUE #3052: Add support enum type
**Description:**
Add support to enum types in serdes standard library

**Describe your task(s)**

**Related Issues (optional):**

**Suggested Labels (optional):**

**Suggested Assignees (optional):**

### PR DESCRIPTION
**Pull request by MohamedSabthar**
## Purpose

Fixes: [#3052](https://github.com/ballerina-platform/ballerina-standard-library/issues/3052)

## Checklist
- [x] Linked to an issue
- [x] Updated the changelog
- [x] Added tests

### CODE REVIEW

File: ballerina/tests/enum_type_test.bal [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+type STATE OPEN|CLOSE;
+
+type RecordWithEnum record {
+    Color color;
+};
+
+type MapWithEnum map<Color>;
+
+@test:Config {}
+public isolated function testBasicEnumType()returns error? {
```
Comment (ThisaruGuruge) [inline ballerina/tests/enum_type_test.bal]:
```suggestion
public isolated function testBasicEnumType() returns error? {
```

File: ballerina/tests/enum_type_test.bal [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+enum Color {
+    RED,
+    GREEN,
+    BLUE
+}
+
+const OPEN = "open";
+const CLOSE = "close";
+type STATE OPEN|CLOSE;
```
Comment (ThisaruGuruge) [inline ballerina/tests/enum_type_test.bal]:
```suggestion
type State OPEN|CLOSE;
```

How does something like this work?

```ballerina
type State OPEN|CLOSE;

type OptionalState OPEN|CLOSE?
```

We should consider this for the other way of defining enums as well.

Comment (MohamedSabthar) [reply ballerina/tests/enum_type_test.bal]: Yes added support for this as well, will update with a test case.

Comment (MohamedSabthar) [reply ballerina/tests/enum_type_test.bal]: Added.

File: ballerina/tests/enum_type_test.bal (hunk 1 above)
Comment (MohamedSabthar) [reply ballerina/tests/enum_type_test.bal]: updated.

**Review by ThisaruGuruge — APPROVED** [review]

**Pull request merged**